dotCMS/src/main/java/com/dotcms/graphql/business/ContentAPIGraphQLTypesProvider.java

```diff
@@ -0,0 +1,245 @@
+
+    private final Map<String, GraphQLType> typesMap = new HashMap<>();
+
+    {
+        // custom type mappings
+        this.fieldClassGraphqlTypeMap.put(BinaryField.class, CustomFieldType.BINARY.getType());
+        this.fieldClassGraphqlTypeMap
+                .put(CategoryField.class, list(CustomFieldType.CATEGORY.getType()));
+        this.fieldClassGraphqlTypeMap.put(ImageField.class, CustomFieldType.FILEASSET.getType());
+        this.fieldClassGraphqlTypeMap.put(FileField.class, CustomFieldType.FILEASSET.getType());
+        this.fieldClassGraphqlTypeMap
+                .put(KeyValueField.class, list(CustomFieldType.KEY_VALUE.getType()));
+        this.fieldClassGraphqlTypeMap.put(CheckboxField.class, list(GraphQLString));
+        this.fieldClassGraphqlTypeMap.put(MultiSelectField.class, list(GraphQLString));
+        this.fieldClassGraphqlTypeMap.put(TagField.class, list(GraphQLString));
+        this.fieldClassGraphqlTypeMap
+                .put(HostFolderField.class, CustomFieldType.SITE_OR_FOLDER.getType());
+
+        // custom data fetchers
+        this.fieldClassGraphqlDataFetcher.put(BinaryField.class, new BinaryFieldDataFetcher());
+        this.fieldClassGraphqlDataFetcher.put(CategoryField.class, new CategoryFieldDataFetcher());
+        this.fieldClassGraphqlDataFetcher.put(ImageField.class, new FileFieldDataFetcher());
+        this.fieldClassGraphqlDataFetcher.put(FileField.class, new FileFieldDataFetcher());
+        this.fieldClassGraphqlDataFetcher.put(KeyValueField.class, new KeyValueFieldDataFetcher());
+        this.fieldClassGraphqlDataFetcher
+                .put(CheckboxField.class, new MultiValueFieldDataFetcher());
+        this.fieldClassGraphqlDataFetcher
+                .put(MultiSelectField.class, new MultiValueFieldDataFetcher());
+        this.fieldClassGraphqlDataFetcher.put(TagField.class, new TagsFieldDataFetcher());
+        this.fieldClassGraphqlDataFetcher
+                .put(HostFolderField.class, new SiteOrFolderFieldDataFetcher());
+    }
+
+    @Override
+    public Collection<GraphQLType> getTypes() throws DotDataException {
+        // we want to generate them always - no cache
+        getContentAPITypes().forEach((graphQLType)->
+                typesMap.put(graphQLType.getName(), graphQLType));
+
+        return typesMap.values();
+    }
+
+    Map<String, GraphQLType> getCachedTypesAsMap() throws DotDataException {
+        if (!UtilMethods.isSet(typesMap)) {
+            getContentAPITypes().forEach((graphQLType)->
+                    typesMap.put(graphQLType.getName(), graphQLType));
+        }
+        return typesMap;
+    }
+
+    private Set<GraphQLType> getContentAPITypes() throws DotDataException {
+
+        Set<GraphQLType> contentAPITypes = new HashSet<>(InterfaceType.valuesAsSet());
+
+        contentAPITypes.add(ExtendedScalars.DateTime);
+
+        List<ContentType> allTypes = APILocator.getContentTypeAPI(APILocator.systemUser())
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable 'allTypes' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5642169)